File: src/7장/7-5.py
galho = input() 
stack = [] 
answer = 0 
for i in range(len(galho)):
   if galho[i] == '(': 
      stack.append(galho[i]) 
   else: 
      if galho[i-1] == '(':
        stack.pop() 
        answer += len(stack) 
      else:
        stack.pop() 
        answer += 1 
print(answer)

File: src/7장/7-5.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		String galho = sc.nextLine();
		Stack<Character> stack = new Stack<>();
		int answer = 0;

		for (int i = 0; i < galho.length(); i++) {
			if (galho.charAt(i) == '(') { 
				stack.push('(');
				continue;
			}
			if (galho.charAt(i) == ')') { 
				stack.pop(); 

				if (galho.charAt(i - 1) == '(') { 
					answer += stack.size(); 
				} else { 
					answer++; 
				}
			}
		}
		System.out.println(answer);
	}
}